Mathlib/SetTheory/Nimber/Field.lean

```diff
@@ -218,4 +222,124 @@ instance : CommRing Nimber where
 instance : IsDomain Nimber where
 instance : CancelMonoidWithZero Nimber where
 
+/-! ### Nimber division -/
+
+mutual
+
+/-- The nimber inverse `a⁻¹` is mutually recursively defined as the smallest nimber not in the set
+`s = invSet a`, which itself is defined as the smallest set with `0 ∈ s` and
+`(1 + (a + a') * b) / a' ∈ s` for `0 < a' < a` and `b ∈ s`.
+
+This preliminary definition "accidentally" satisfies `invAux 0 = 1`, which the real inverse
+corrects. The lemma `inv_eq_invAux` can be used to transfer between the two. -/
+def invAux (a : Nimber) : Nimber :=
+  sInf (invSet a)ᶜ
+termination_by (a, 1)
+
+/-- The set in the definition of `invAux a`. -/
+def invSet (a : Nimber) : Set Nimber :=
+  ⋂₀ {s | 0 ∈ s ∧ ∀ a' < a, a' ≠ 0 → ∀ b ∈ s, invAux a' * (1 + (a + a') * b) ∈ s}
+termination_by (a, 0)
```

Comment:
Can't these two defs be made `private`? Do you have plans to use them in later files?

Reply:
See my comment below, I don't have plans to use them but I do think they're still worth making public.